src/odemis/driver/rigol.py

```diff
@@ -259,56 +256,59 @@ def sendOrderCommand(self, cmd, val=""):
         if not self._is_connected:
             raise IOError("Device %s not connected." % (self._host,))
 
-        msg = "%s %s\n" % (cmd, val)
-
+        msg = ("%s %s\n" % (cmd, val)).encode('acii')
+        logging.debug("Sending command %s", msg.decode('latin1', 'backslashreplace'))
         with self._net_access:
             self.socket.sendall(msg)
 
     def sendQueryCommand(self, cmd):
         """
         Sends one command, and expect a reply
         cmd (str): command to send, including the ?
-
+        returns:
+            ans (str): response of the driver
         raises:
             IOError: if problem with sending/receiving data over the connection
         """
         if not self._is_connected:
             raise IOError("Device %s not connected." % (self._host,))
 
-        msg = "%s\n" % cmd
+        msg = ("%s\n" % cmd).encode('ascii')
+        logging.debug("Sending command %s", msg.decode('latin1', 'backslashreplace'))
 
         with self._net_access:
             self.socket.sendall(msg)
 
             # read the answer
             end_time = time.time() + 0.5
-            ans = ""
+            ans = b""
             while True:
                 try:
                     data = self.socket.recv(4096)
                 except socket.timeout:
                     raise IOError("Controller %s timed out after %s" %
-                                  (self._host, msg.encode('string_escape')))
+                                  (self._host, msg.decode('latin1', 'backslashreplace')))
 
                 if not data:
                     logging.debug("Received empty message")
 
                 ans += data
                 # does it look like we received a full answer?
-                if "\n" in ans:
+                if b"\n" in ans:
                     break
 
                 if time.time() > end_time:
                     raise IOError("Controller %s timed out after %s" %
-                                  (self._host, msg.encode('string_escape')))
+                                  (self._host, msg.decode('latin1', 'backslashreplace')))
                 time.sleep(0.01)
 
-        logging.debug("Received: %s", ans.encode('string_escape'))
+        logging.debug("Received: %s", ans.decode('latin1', 'backslashreplace'))
 
-        ans, left = ans.split("\n", 1)  # remove the end of line characters
+        ans, left = ans.split(b"\n", 1)  # remove the end of line characters
         if left:
             logging.error("Received too much data, will discard the end: %s",
-                          left.encode('string_escape'))
+                          left.decode('latin1', 'backslashreplace'))
+        ans = ans.decode('latin1') if sys.version_info[0] >= 3 else ans
```

Comment:
Please, drop the `if`. On Python 2 it's a no-op, and that's fine.
(and then you can remove the `import sys`)

Reply:
I think it is necessary.
`decode` should return a unicode in python2 and not a str.
`isinstance(u'a', str)` will return False